extensions/typescript-language-features/src/utils/electron.ts

```diff
@@ -22,8 +22,21 @@ const getRootTempDir = (() => {
 	};
 })();
 
+export const getInstanceDir = (() => {
+	let dir: string | undefined;
+	return () => {
```

Comment:
why does this return a function?

Reply:
This is so that every time we call `getInstanceDir()` we do not reinitialize the dir variable after each run. The dir variable is initialized once upon self execution of the outside function and then isn't reinitialized on subsequent calls because the getInstanceDir only holds the returned function.